Gems/Atom/Tools/MaterialCanvas/Code/Source/Document/MaterialGraphCompiler.cpp

```diff
@@ -422,9 +424,11 @@ namespace MaterialCanvas
         switch (score)
         {
         case 4:
-            return ConvertToVector<AZ::Vector4>(slotValue);
+            // Skipping color to vector conversions that they export as the correct type with the material type.
+            return slotValue.is<AZ::Color>() ? slotValue : ConvertToVector<AZ::Vector4>(slotValue);
         case 3:
-            return ConvertToVector<AZ::Vector3>(slotValue);
+            // Skipping color to vector conversions that they export as the correct type with the material type.
+            return slotValue.is<AZ::Color>() ? slotValue : ConvertToVector<AZ::Vector3>(slotValue);
```

Comment:
nit: The comments reads strangely, is it supposed to be "...so that they..." ?

Reply:
Comments updated